cli/src/lib/plugin/start.js

```diff
@@ -0,0 +1,47 @@
+// Based on CRA start script
+
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.BABEL_ENV = 'development'
+process.env.NODE_ENV = 'development'
+
+const getPublicUrlOrPath = require('react-dev-utils/getPublicUrlOrPath')
+const webpack = require('webpack')
+const WebpackDevServer = require('webpack-dev-server')
+const webpackConfigFactory = require('./webpack.config')
+
+module.exports = async ({ port, config, paths }) => {
+    const webpackConfig = webpackConfigFactory({
+        // todo: change to development, but this creates a compilation error
```

Comment:
Can you explain why that would cause a compilation error?  What are the downsides of building for production here?

Reply:
It throws the error in Figure 1, which seems to be a conflict with the dev server for both the app and the plugin writing files to the same place? I didn't look into it much because it works in `'production'` mode -- the downsides are quality of life things:
1. It’s harder to debug errors in the plugin because the files are minified and the source maps aren’t working (Figure 2)
2. There’s more webpack chatter in the terminal as it rebuilds for every code change (Figure 3)
3. I’m guessing the plugin has a bit slower start and hot reload times due to the extra compilation steps

Figure 1: `'development'` mode error
![plugin-start-fig-1](https://user-images.githubusercontent.com/49666798/203808538-d741a25d-453e-4782-8a67-46a3754f1226.png)

Figure 2: difficult debugging
![plugin-start-fig-2](https://user-images.githubusercontent.com/49666798/203808548-2acb5856-08b7-40d5-a301-c20296440342.png)

Figure 3: webpack chatter
![plugin-start-fig-3](https://user-images.githubusercontent.com/49666798/203808556-d5429c2f-ff07-4e21-8906-b59e8835ebf9.png)